### DISCUSSION TITLE
ICs From CSV

### DISCUSSION THREAD

**Discussion by gj19866** [idaholab/moose#25638] [Q&A General] [answered]
Hi there, 
Still new to MOOSE and getting used to what it has built-in to help!

I am trying to set up the initial conditions of a variable from a CSV.
Is there a 'best' way to do this?

I am currently using the `PropertyReadFile` user object, to write a function using `PiecewiseConstantFromCSV`, and then writing that to an IC using `FunctionIC`.

I'm sure there must be a better way to do this!

For reference the reason I think I need to use a CSV is because there are two variables, x and y, and x^2+y^2=1, and I want x to be random between -1 to 1, so I have written a python script to write a `.csv`. 

It would also be cool to know if I could use something else in MOOSE to solve the whole problem!

Thanks,
Gillian

**Comment by GiudGiud** [comment, marked as answer]
Hello

You can also write your IC object directly in C++. You can either read a CSV directly there or even generate the initial condition there too

The 3 objects route you are using right now is fine. It's just that way to make sure that only one object has to read the CSV file.

Guillaume

**Reply by gj19866** [reply]
Òh amazing! In that case I'll leave it how it is for now, then I can always optimise later!

Thank you Guillamue,

Gillian
